frontend/src/stores/user.js

```diff
@@ -76,10 +76,25 @@ class userStore {
         this.balance = userData.balance;
         this.phoneNumber = userData.phoneNumber;
     }
+    resetCredentials() {
+        this.token = null;
+        this.username = null;
+        this.balance = null;
+        this.phoneNumber = null;
+    }
     setBalance(amount) {
         if (amount > this.balance) throw new Error('Insufficient fund!');
         else this.balance = this.balance - amount;
     }
+
+    logout() {
+        try {
+            logoutApi();
+            this.resetCredentials();
+        } catch (error) {
```

Comment:
In which case would this catch block be triggered in the current implementation?

Reply:
yes, because logout Api will throw if there is error or server rejects

Reply:
I don't think it will be caught in the catch block since `await` is not used and promises are asynchronous.